src/query.js

```diff
@@ -18,13 +19,15 @@ class Query {
    *
    * @param {DHT} dht - DHT instance
    * @param {Buffer} key
-   * @param {function(PeerId, function(Error, Object))} query - The query function to exectue
+   * @param {function(PeerId, Object, function(Error, Object))} query - The query function to exectue
```

Comment:
Please add an explanation for the Object state

Typo: `exectue`

Reply:
Sure

Reply:
New version has better types.